GVRf/Framework/framework/src/main/jni/objects/mesh.cpp

```diff
@@ -32,8 +32,18 @@
 
 namespace gvr {
 Mesh* Mesh::createBoundingBox() {
+    if (bounding_box_mesh_ != nullptr && have_bounding_volume_) {
+        return bounding_box_mesh_;
```

Comment:
There is code that expects createBoundingBox to return a new instance always. The Java side does that (see GVRMesh.getBoundingBox); Java assumes it owns the native ptr and will delete it at some point. There is also Picker::pickSceneObjectAgainstBoundingBox. You might have to create a special version of this method that does what you want in the cases where it matters for your change.